### PR TITLE
Bug/pagination

### LINKED ISSUE #81: Pagination not working with pageSize less than offset default of 10
**Describe the bug**
The pagination seem to be omitting pageSize and generating negetive values. 

**To Reproduce**
```tsx
<Pagination
    value={paginationValue}
    onChange={setPagination}
    size={3}
    useFirstAndLast={true}
/>
```

**Screenshots**
image

**Additional context**
The `getlist ` should be calculated once and only when there is a change in value.

### PR DESCRIPTION
**Pull request by DanSallau**
fixes #81

### CODE REVIEW

File: src/Pagination/Pagination.tsx [hunk 1]
```diff
@@ -33,27 +35,37 @@ export const Pagination: React.FunctionComponent<PaginationProps> = React.memo((
      * @param {number} offset The offset to show the numbers
```
Comment (mario-subo) [inline src/Pagination/Pagination.tsx]: JSdocs need to reflect the changes

File: src/Pagination/Pagination.tsx [hunk 2]
```diff
@@ -33,27 +35,37 @@ export const Pagination: React.FunctionComponent<PaginationProps> = React.memo((
...
-                list.push(i);
-            }
-        } else {
-            range.min = value + 1 - (Math.ceil(offset / 2));
-            range.max = range.min + offset - 1;
-            for (let i = range.min; i <= range.max; i++) {
-                list.push(i);
+
+        // generate the array
+        for (let i = 1; i <= pagingSize; i++) {
```
Comment (mario-subo) [inline src/Pagination/Pagination.tsx]: For consistency consider deciding if you want to either send all outside values into this function as a param or just use the values directly.  Right now you're sending the `value` and `length` as params but `pagingSize` is being used directly. 

You can also consider building the list and setting it to an internal state within a React.useEffect which would run every time any of the values you need change. Then you would just render the list within the return. Either way you should clean it up a bit.

**Comment by mario-subo** [comment]
I think there's also a bug with the next button where the icon doesn't turn white on hover, can you check it out?

File: src/Pagination/Pagination.tsx [hunk 3]
```diff
@@ -81,7 +104,7 @@ export const Pagination: React.FunctionComponent<PaginationProps> = React.memo((
                                 </button>
                             </li>
                         }
-                        {getList(props.value, props.size, Initialoffset).map((num) => {
+                        {list.map((num) => {
```
Comment (yousifalraheem) [inline src/Pagination/Pagination.tsx]:
```suggestion
                        {list.map((num: number) => {
```

File: src/Pagination/Pagination.tsx [hunk 4]
```diff
@@ -126,7 +149,7 @@ export const Pagination: React.FunctionComponent<PaginationProps> = React.memo((
                 }
                 {props.useDotNav &&
                     <ul className={"pagination dotnav"}>
-                        {getList(props.value, props.size, props.size).map((num) => {
+                        {list.map((num) => {
```
Comment (yousifalraheem) [inline src/Pagination/Pagination.tsx]:
```suggestion
                        {list.map((num: number) => {
```

File: src/Pagination/readme.md [hunk 5]
```diff
@@ -44,6 +44,5 @@ These are the current available properties:
 | useFirstAndLast? | `boolean`                 | Use first and last navigation buttons (default: `false`) |
 | useTextNav?      | `boolean`                 | Use text-base navigation buttons (default: `false`)      |
 | value            | `number`                  | Current page                                             |
+| pagingLength?    | `number`                  | The length of the pagination or number of pages          |
 
-## Footnote
-1. Dot navigation does not support `offset`, it is not meant to have big size navigation. For that, use the normal numbered navigation.
```
Comment (yousifalraheem) [inline src/Pagination/readme.md]: I think you need to put back this footnote since it's important

File: src/Pagination/Pagination.tsx [hunk 6]
```diff
@@ -23,37 +24,60 @@ export interface PaginationProps {
...
-            range.max = size;
-            for (let i = range.min; i <= range.max; i++) {
-                list.push(i);
-            }
-        } else {
-            range.min = value + 1 - (Math.ceil(offset / 2));
-            range.max = range.min + offset - 1;
-            for (let i = range.min; i <= range.max; i++) {
-                list.push(i);
+    function getList(value: number, length: number): void {
```
Comment (yousifalraheem) [inline src/Pagination/Pagination.tsx]: Don't you think a method called `getList` should return something? if not, then it shouldn't be called `getList` just to not confuse the reader

File: src/Pagination/Pagination.tsx [hunk 7]
```diff
@@ -33,27 +35,37 @@ export const Pagination: React.FunctionComponent<PaginationProps> = React.memo((
...
-            for (let i = range.min; i <= range.max; i++) {
-                list.push(i);
+
+        // generate the array
+        for (let i = 1; i <= pagingSize; i++) {
+            list.push(i);
+        }
+
+        // the median value is what you use to obtain the centre of the page or equilibrium
+        const medianValue = Math.ceil(length / 2);
```
[the remark this answers is not in the document]
Comment (yousifalraheem) [reply src/Pagination/Pagination.tsx]: Why is this maked as resolved? The type still haven't been added

**Comment by yousifalraheem** [comment]
Can you add more unit test? **Coveralls** is noticing a coverage decrease, this should be easily fixed

File: src/Pagination/Pagination.tsx [hunk 8]
```diff
@@ -23,37 +24,59 @@ export interface PaginationProps {
...
-    const Initialoffset: number = props.offset ? props.offset : 10;
+
+    const [list, setList] = React.useState<Array<number>>([]);
+    const [dotnavList, setDotnavList] = React.useState<Array<number>>([]);
+    const [pagingSize, setPagingSize] = React.useState<number>(0);
+
+    React.useEffect(() => {
+        const initialOffset: number = props.offset ? props.offset : 10;
+        setPagingSize(Math.ceil(props.size / initialOffset));
+    }, [props.offset, props.pagingLength, props.size]);
```
Comment (omarVodiaK) [inline src/Pagination/Pagination.tsx]: is props.pagingLength required here, the effect would run if pagingLength changes.

File: src/Pagination/Pagination.tsx [hunk 9]
```diff
@@ -23,37 +24,59 @@ export interface PaginationProps {
...
+
+    React.useEffect(() => {
+        const intialLength: number = props.pagingLength ? props.pagingLength : 5;
+        generateList(props.value, intialLength);
+    }, [pagingSize, props.value]);
```
Comment (omarVodiaK) [inline src/Pagination/Pagination.tsx]: maybe props.pagingLength should be added here.

**Comment by omarVodiaK** [comment]
You should update the branch by merging latest develop

File: src/Pagination/Pagination.tsx (hunk 8 above)
Comment (yousifalraheem) [reply src/Pagination/Pagination.tsx]:
```suggestion
    }, [props.offset, props.size]);
```

File: src/Pagination/Pagination.tsx (hunk 9 above)
Comment (yousifalraheem) [reply src/Pagination/Pagination.tsx]:
```suggestion
    }, [pagingSize, props.value, props.pagingLength]);
```

File: src/Pagination/Pagination.tsx (hunk 8 above)
Comment (DanSallau) [reply src/Pagination/Pagination.tsx]: nice catch, thanks.

**Pull request merged**